lib/fibonacci.rb

```diff
@@ -1,8 +1,20 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n), where n is the input number to find
+# Space Complexity - O(n), where n is the input number to find
 # Hint, you may want a recursive helper method
-def fibonacci(n)
```

Comment:
This works, but `memos` is going to grow to `n` length.  However to find fib(n) you only need to know fib(n-1) and fib(n-2), so can you solve this and only remember the last two numbers in the sequence.